roles/conduit/files/conduit_leds.py

```diff
@@ -0,0 +1,507 @@
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+# cat /sys/devices/platform/mts-io/led-{a, b, c, d} aka {cd, sig1, sig2, sig3}
+
+import argparse
+from contextlib import contextmanager
+import errno
+import fcntl
+import logging
+from logging.handlers import SysLogHandler
+import os
+import pprint
+import psutil
+import re
+import socket
+import stat
+import struct
+import subprocess
+import sys
+import time
+
+class LockFileTimeout(Exception):
+    def __init__(self, error):
+        self.value = error
+    def __str__(self):
+         return repr(self.value)
+
+@contextmanager
+def pidfilelock(name):
+    """ Context to lock a pid file """
+
+    time_left = 30
+    pidfile_path = os.path.join("/var/run", name + ".pid")
+    lock_file = open(pidfile_path, 'w+')
+    while True:
+        try:
+            logging.debug("Attempting to lock %s", pidfile_path)
+            fcntl.flock(lock_file, fcntl.LOCK_EX | fcntl.LOCK_NB)
+            lock_file.write(str(os.getpid()) + '\n')
+            lock_file.flush()
+            logging.debug("Wrote %d to %s", os.getpid(), pidfile_path)
+            break
+        except IOError as err:
+            if err.errno != errno.EAGAIN:
+                raise err
+            else:
+                logging.debug("Timeout trying to lock", pidfile_path)
+                time.sleep(1)
+                time_left -= 1
+                if time_left == 0:
+                    raise LockFileTimeout("Unable to lock %s" % pidfile_path)
+
+    try:
+        yield lock_file
+    finally:
+        logging.debug("Unlocking %s", pidfile_path)
+        fcntl.flock(lock_file, fcntl.LOCK_UN)
+        os.unlink(pidfile_path)
+        lock_file.close()
+               
+class Defaults(object):
+    """ Read a /etc/defaults file """
+
+    ROOT = "/etc/default"
+    COMMENT_RE=re.compile(r'\s*#.*')
+    ASSIGN_RE=re.compile(r'\s*(?P<name>[A-Z][A-Z_0-9]+)=(?P<value>[^#]*)')
+
+    def __init__(self, name):
+        self._name = name
+        self._path = os.path.join(self.ROOT, self._name)
+        self.vars = {}
+
+        with open(self._path, "r") as fp:
+            for line in fp:
+                line = line.strip()
+                if len(line) == 0 or self.COMMENT_RE.match(line):
+                    continue
+                logging.debug("Defaults: %s: %s", self._path, line)
+                match = self.ASSIGN_RE.match(line)
+                if not match:
+                    continue
+                value = match.group('value')
+                try:
+                    value = int(value)
+                except ValueError:
+                    pass
+                try:
+                    for quote in ["'", '"']:
+                        if value.startswith(quote) and value.endswith(quote):
+                            value = value[1:-1]
```

Comment:
I think this will cause a value of the form `'"duck"' `to be converted to `duck`. Should you `break` after line 114?

Reply:
ducking-a.

Fixed